02_bubble/jacklee/main_test.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
+	//Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	//when
+	main()
+
+	//Then
+	expected := "[1 2 3 5]"
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output")
+}
+
+var testCases = []struct {
+	in  []int
+	out []int
+}{
```

Comment:
This one is largely a matter of personal preference, but in this case I would prefer a one-liner:
```go
var testCases = []struct{in, out []int}{
	…
}
```